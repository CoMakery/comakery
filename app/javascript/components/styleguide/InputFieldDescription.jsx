import React from 'react'
import classNames from 'classnames'
import InputField from './InputField'

class InputFieldDescription extends React.Component {
  render() {
    const {
      className,
      ...other
    } = this.props

    const classnames = classNames(
      'input-field__description',
      className
    )

    return (
      <React.Fragment>
        <InputField
          className={classnames}
          type="textarea"
          {...other}
        />
      </React.Fragment>
    )
  }
}

export default InputFieldDescription
