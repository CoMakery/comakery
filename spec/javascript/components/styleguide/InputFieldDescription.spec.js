import React from 'react'
import { shallow, mount, render } from 'enzyme'
import InputFieldDescription from 'components/styleguide/InputFieldDescription'

describe('InputFieldDescription', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<InputFieldDescription />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.input-field__description')).toBe(true)
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<InputFieldDescription className="__test" />)

    expect(wrapper.exists('.input-field__description.__test')).toBe(true)
  })
})
