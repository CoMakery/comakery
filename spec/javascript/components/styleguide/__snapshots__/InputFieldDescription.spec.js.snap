// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFieldDescription renders correctly without props 1`] = `
<Fragment>
  <InputField
    checkboxText=""
    checked={false}
    className="input-field__description"
    copyOnClick={false}
    disabled={false}
    errors={Array []}
    eventHandler={[Function]}
    imgInputRef={null}
    imgPreviewDimensions="40x40"
    imgPreviewUrl=""
    imgRequirements=""
    imgVerifier={[Function]}
    max=""
    min=""
    name="field"
    pattern=".*"
    placeholder="Please enter value"
    readOnly={false}
    recommended={false}
    required={false}
    selectEntries={Array []}
    step=""
    symbolLimit={100}
    title="title"
    type="textarea"
    value=""
  />
</Fragment>
`;
